File: scratch/debug_mohammed.js
const Database = require('better-sqlite3');
const db = new Database('C:\\Users\\Orbix Soft. Solution\\AppData\\Roaming\\laundry-box\\laundry_pos.sqlite');

try {
  const customer = db.prepare("SELECT * FROM customers WHERE phone = '+971588851680'").get();
  console.log("CUSTOMER:", customer);

  if (customer) {
    const custId = customer.id;
    console.log("\n--- ORDERS ---");
    console.log(db.prepare("SELECT * FROM orders WHERE customerId = ?").all(custId));

    console.log("\n--- DELETED ORDERS ---");
    console.log(db.prepare("SELECT * FROM deleted_orders WHERE customerId = ?").all(custId));

    console.log("\n--- PAYMENTS ---");
    console.log(db.prepare("SELECT * FROM payments WHERE customerId = ?").all(custId));

    console.log("\n--- ADVANCE ALLOCATIONS ---");
    console.log(db.prepare("SELECT * FROM advance_allocations WHERE paymentId IN (SELECT id FROM payments WHERE customerId = ?)").all(custId));
  }
} catch (e) {
  console.error(e);
}

File: scratch/debug_mohammed.py
import sqlite3
import json

db_path = 'C:\\Users\\Orbix Soft. Solution\\AppData\\Roaming\\laundry-box\\laundry_pos.sqlite'
conn = sqlite3.connect(db_path)
conn.row_factory = sqlite3.Row
cursor = conn.cursor()

try:
    cursor.execute("SELECT * FROM customers WHERE phone = '+971588851680'")
    customer = cursor.fetchone()
    if customer:
        cust_id = customer['id']
        print("CUSTOMER:", dict(customer))

        print("\n--- ORDERS ---")
        cursor.execute("SELECT * FROM orders WHERE customerId = ?", (cust_id,))
        for r in cursor.fetchall():
            print(dict(r))

        print("\n--- DELETED ORDERS ---")
        cursor.execute("SELECT * FROM deleted_orders WHERE customerId = ?", (cust_id,))
        for r in cursor.fetchall():
            print(dict(r))

        print("\n--- PAYMENTS ---")
        cursor.execute("SELECT * FROM payments WHERE customerId = ?", (cust_id,))
        for r in cursor.fetchall():
            print(dict(r))

        print("\n--- ADVANCE ALLOCATIONS ---")
        cursor.execute("""
            SELECT a.* FROM advance_allocations a
            JOIN payments p ON a.paymentId = p.id
            WHERE p.customerId = ?
        """, (cust_id,))
        for r in cursor.fetchall():
            print(dict(r))
except Exception as e:
    print(e)
finally:
    conn.close()
